sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -114,21 +116,20 @@ def _csr_mean_variance_axis0(cnp.ndarray[floating, ndim=1, mode="c"] X_data,
         integral i, col_ind
         cnp.float64_t diff
         # means[j] contains the mean of feature j
-        cnp.ndarray[cnp.float64_t, ndim=1] means = np.zeros(n_features)
+        cnp.float64_t[:] means = np.zeros(n_features)
```

Comment:
Specifying contiguity when possible is better. Can you specify contiguity for memoryview using `::1`?

For instance, here:
```suggestion
        cnp.float64_t[::1] means = np.zeros(n_features)
```

This suggestion also applies in other places.

For more information, see [the dedicated subsection in Cython's documentation](https://cython.readthedocs.io/en/latest/src/userguide/memoryviews.html#specifying-more-general-memory-layouts).

Reply:
Should I update all the memory views in this file to be contiguous? Or is there a specific criteria to judge whether doing this is useful in certain cases but not in others?

Reply:
> Should I update all the memory views in this file to be contiguous?

All the ones in the implementations must be declared as being contiguous, yes.

> Or is there a specific criteria to judge whether doing this is useful in certain cases but not in others?

For proper support of variability of inputs arrays, I would not modify the signatures' memoryviews' declarations: not declaring them as not being contiguous leaves room for flexibility.

For similar previous discussions, see https://github.com/scikit-learn/scikit-learn/pull/25323.

Reply:
Thanks!

Reply:
You are welcome, Omar. :)